query_modules/example.cpp

```diff
@@ -18,7 +18,11 @@ void ProcImpl(std::vector<mgp::Value> arguments, mgp::Graph graph, mgp::RecordFa
 
 void SampleReadProc(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
   try {
-    mgp::memory = memory;
+    // The outcommented way of assigning the memory pointer is still
+    // working, but it is deprecated because of certain concurrency
+    // issues. Please use the guard instead.
+    // mgp::memory = memory;
+    mgp::MemoryDispatcherGuard guard(memory);
```

Comment:
I think we have quite a few query modules now implemented in MAGE so I guess we also need to make an additional PR to make these concurrently executable as well?

Reply:
this seems cool though, no overhead at all

Reply:
Yes, the MAGE modules will have to be modified as well, we discussed it with @gitbuda and @antoniofilipovic, we will get to it once Fico is back.